src/test/java/com/bizo/awsstubs/services/dynamodb2/AmazonDynamoDBStubTest.java

```diff
@@ -380,11 +379,26 @@ private PutItemResult putItem(String attributeName, String attributeValue) throw
     return result;
   }
 
-  @Test(expected=InternalServerErrorException.class)
+  @Test
   public void test_query() throws Exception {
-    QueryResult result = dynamoDb.query(new QueryRequest());
+    createTable();
+    putItem(TEST_ATTRIBUTE, TEST_ATTRIBUTE_VALUE);
 
-    return;
+    Condition keyCondition = new Condition()
+      .withComparisonOperator(ComparisonOperator.EQ)
+      .withAttributeValueList(new AttributeValue().withS(TEST_ATTRIBUTE_VALUE));
+
+    Map<String, Condition> keyConditions = new HashMap<String, Condition>();
+    keyConditions.put(TEST_ATTRIBUTE, keyCondition);
+
+    QueryRequest queryRequest = new QueryRequest()
+      .withTableName(TEST_TABLE_NAME)
+      .withKeyConditions(keyConditions);
+
+    QueryResult result = dynamoDb.query(queryRequest);
+    Double units = result.getConsumedCapacity().getCapacityUnits();
```

Comment:
Asserting on the capacity units is fine, but should probably also assert against the returned rows/items?